File: 15/1.rs
use std::fs;

fn hash(string: &str) -> u32 {
    string.chars().fold(0, |a, c| ((a + c as u32) * 17) % 256)
}

fn main() {
    let sum: u32 = fs::read_to_string("./15/input.txt")
        .unwrap()
        .split(',')
        .map(|s| hash(s))
        .sum();
    println!("{sum}");
}

File: 2023/4/1.js
const cards = require("node:fs")
  .readFileSync("./input.txt")
  .toString()
  .split("\n")
  .map((line) => {
    const items = line.split(/ \| +| +/g);
    return [new Set(items.slice(2, 12)), new Set(items.slice(12))];
  });

let sum = 0;
for (const [A, B] of cards) {
  const wins = [...B].filter((b) => A.has(b)).length;
  if (wins > 0) sum += 1 << (wins - 1);
}

console.log({ sum });
